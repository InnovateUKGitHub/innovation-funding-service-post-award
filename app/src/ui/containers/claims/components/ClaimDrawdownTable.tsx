import { LoanDto, ProjectDto } from "@framework/dtos";
import { checkProjectCompetition } from "@ui/helpers/check-competition-type";
import { getPending } from "@ui/helpers/get-pending";
import { useStores } from "@ui/redux";

import * as ACC from "@ui/components";

type ProjectOptions = Pick<ProjectDto, "id" | "competitionType">;

export interface ClaimDrawdownTableProps extends ProjectOptions {
  /** @description Current project period number could be ahead/behind of the claim period  */
  requiredPeriod: number;
}

function useClaimDrawdown(props: ClaimDrawdownTableProps) {
  const { isLoans } = checkProjectCompetition(props.competitionType);
  const stores = useStores();

  // Note: Only drawdowns can display drawdown data
  if (!isLoans) return null;

  const loanQuery = stores.loans.get(props.id, undefined, props.requiredPeriod);
  const { error, ...rest } = getPending(loanQuery);

  if (error?.message === "No loan found.") return null;

  if (rest.isRejected) throw Error(error?.message ?? "There was an error fetching data within ClaimDrawdownTable");

  // Note: We require total values for the UI
  if (rest.payload && rest.payload.totals === undefined) throw Error("Loan totals must be available.");

  const isLoading = !rest.payload || rest.isLoading;

  return {
    ...rest,
    isLoading,
  };
}

export function ClaimDrawdownTable(props: ClaimDrawdownTableProps) {
  const claimDrawdown = useClaimDrawdown(props);

  if (!claimDrawdown) return null;

  if (claimDrawdown.isLoading) return <ACC.LoadingMessage />;

  const PeriodDrawdownTable = ACC.TypedTable<Required<LoanDto>>();

  return (
    <PeriodDrawdownTable.Table qa="period-loan-table" data={[claimDrawdown.payload as Required<LoanDto>]}>
      <PeriodDrawdownTable.Custom header="Current drawdown" qa="drawdown-id" value={x => x.period} />

      <PeriodDrawdownTable.Currency header="Total loan amount" qa="drawdown-amount" value={x => x.totals.totalLoan} />

      <PeriodDrawdownTable.Currency
        header="Drawdowns to date"
        qa="drawdown-totals-todate"
        value={x => x.totals.totalPaidToDate}
      />

      <PeriodDrawdownTable.Currency header="Drawdown this period" qa="drawdown-current-amount" value={x => x.amount} />

      <PeriodDrawdownTable.Currency
        header="Remaining loan amount"
        qa="drawdown-remaining-amount"
        value={x => x.totals.remainingLoan}
      />
    </PeriodDrawdownTable.Table>
  );
}
