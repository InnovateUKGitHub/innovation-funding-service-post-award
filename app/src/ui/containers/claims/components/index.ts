export * from "./ClaimsDashboardGuidance";
export * from "./EnumDocuments";
export * from "./allowed-documents.hook";
