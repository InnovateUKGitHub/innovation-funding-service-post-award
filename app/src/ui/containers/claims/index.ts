export * from "./dashboard";
export * from "./details";
