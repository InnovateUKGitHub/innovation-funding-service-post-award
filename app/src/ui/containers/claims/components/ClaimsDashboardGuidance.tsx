import { PartnerDto } from "@framework/dtos";
import { getAuthRoles, ProjectRole } from "@framework/types";
import { checkProjectCompetition } from "@ui/helpers/check-competition-type";
import { useContent } from "@ui/hooks";

import * as Acc from "@ui/components";
import { Content, EmailContent } from "@ui/components";

export interface ClaimsDashboardGuidanceProps {
  overdueProject: PartnerDto["overdueProject"];
  competitionType?: PartnerDto["competitionType"];
  roles?: PartnerDto["roles"];
}

export const ClaimsDashboardGuidance = ({
  overdueProject,
  competitionType = "",
  roles = ProjectRole.Unknown,
}: ClaimsDashboardGuidanceProps) => {
  const { getContent } = useContent();

  let guidanceMessage: React.ReactElement;

  if (overdueProject) {
    guidanceMessage = (
      <Content
        value={x => x.claimsMessages.overdueGuidanceMessage.message}
        components={[<EmailContent key="email" value={x => x.claimsMessages.overdueGuidanceMessage.email} />]}
      />
    );
  } else {
    const isPartnerFc = getAuthRoles(roles).isFc;
    const { isCombinationOfSBRI } = checkProjectCompetition(competitionType);

    const defaultGuidanceMessage = getContent(x => x.claimsMessages.guidanceMessage);

    const displaySbriGuidance = isPartnerFc && isCombinationOfSBRI;

    if (displaySbriGuidance) {
      return <Acc.Renderers.SimpleString>{defaultGuidanceMessage}</Acc.Renderers.SimpleString>;
    }

    guidanceMessage = <Acc.Renderers.Markdown trusted value={defaultGuidanceMessage} />;
  }

  return <Acc.ValidationMessage qa="guidance-message" messageType="info" message={guidanceMessage} />;
};
