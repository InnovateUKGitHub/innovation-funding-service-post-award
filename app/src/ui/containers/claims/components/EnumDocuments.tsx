import { DocumentDescription } from "@framework/constants/documentDescription";
import { DropdownOption } from "@ui/components/form";
import React from "react";
import { useEnumDocuments } from "./allowed-documents.hook";

interface EnumDocumentsProps {
  documentsToCheck: Readonly<DocumentDescription[]>;
  children: (documents: DropdownOption[]) => React.ReactElement;
}

/**
 * wrapper component to allow use of `useEnumDocuments` hook with class components
 */
export function EnumDocuments({ documentsToCheck, children }: EnumDocumentsProps) {
  const documents = useEnumDocuments(DocumentDescription, documentsToCheck);

  return children(documents);
}
