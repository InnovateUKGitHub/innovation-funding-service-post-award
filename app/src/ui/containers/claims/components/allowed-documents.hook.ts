import { DocumentDescription, getDocumentDescriptionContentSelector } from "@framework/constants";
import { getAllEnumValues } from "@shared/enumHelper";
import { DropdownOption } from "@ui/components";
import { useContent } from "@ui/hooks";

export function useEnumDocuments(
  enumDocuments: object,
  documentsToCheck: Readonly<DocumentDescription[]>,
): DropdownOption[] {
  const { getContent } = useContent();
  const getRawDocs: number[] = getAllEnumValues(enumDocuments);

  return getRawDocs.reduce<DropdownOption[]>((acc, doc) => {
    const isInvalidDocument = !documentsToCheck.includes(doc);

    if (isInvalidDocument) return acc;

    const id = doc.toString();
    const value = getContent(getDocumentDescriptionContentSelector(doc));

    return [
      ...acc,
      {
        id,
        value,
      },
    ];
  }, []);
}
