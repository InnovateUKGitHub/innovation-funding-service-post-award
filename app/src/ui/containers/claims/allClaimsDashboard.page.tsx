import { ClaimDto, PartnerDto, ProjectDto } from "@framework/dtos";
import { BaseProps, defineRoute } from "@ui/containers/containerBase";
import * as Acc from "@ui/components";
import { Pending } from "@shared/pending";
import { DateTime } from "luxon";
import { useStores } from "@ui/redux";
import { getClaimDetailsLinkType } from "@ui/components/claims/claimDetailsLink";
import { getPartnerName } from "@ui/components";
import { checkProjectCompetition } from "@ui/helpers/check-competition-type";
import { roundCurrency } from "@framework/util";
import { ProjectRole, ProjectStatus } from "@framework/constants";
import { IRoutes } from "@ui/routing";
import { getAuthRoles } from "@framework/types";
import { getLeadPartner } from "@framework/util/partnerHelper";
import { useProjectStatus } from "@ui/hooks";
import { useProjectParticipants } from "@ui/features/project-participants";
import { ClaimsDashboardGuidance } from "./components";

export interface AllClaimsDashboardParams {
  projectId: string;
}

interface AllClaimsDashboardData {
  projectDetails: Pending<ProjectDto>;
  partners: Pending<PartnerDto[]>;
  currentClaims: Pending<ClaimDto[]>;
  previousClaims: Pending<ClaimDto[]>;
  routes: IRoutes;
  messages: string[];
}

function AllClaimsDashboardComponent(props: AllClaimsDashboardParams & AllClaimsDashboardData & BaseProps) {
  const { isMultipleParticipants } = useProjectParticipants();

  const { isActive: isProjectActive } = useProjectStatus();

  const combined = Pending.combine({
    projectDetails: props.projectDetails,
    partners: props.partners,
    previousClaims: props.previousClaims,
  });

  const renderContents = (projectDetails: ProjectDto, partners: PartnerDto[], previousClaims: ClaimDto[]) => {
    const { isCombinationOfSBRI } = checkProjectCompetition(projectDetails.competitionType);
    const { isFc } = getAuthRoles(projectDetails.roles);

    const leadPartner = getLeadPartner(partners);
    const isLeadPartnerFc = leadPartner && getAuthRoles(leadPartner.roles).isFc;
    const hasWithdrawnPartners = partners.some(partner => partner.isWithdrawn);

    return (
      <Acc.Page
        pageTitle={<Acc.Projects.Title {...projectDetails} />}
        backLink={<Acc.Projects.ProjectBackLink project={projectDetails} routes={props.routes} />}
        project={projectDetails}
        partner={isLeadPartnerFc ? leadPartner : undefined}
      >
        {isMultipleParticipants && isFc && renderGuidanceMessage(isCombinationOfSBRI, partners)}

        {hasWithdrawnPartners && (
          <Acc.ValidationMessage messageType="info" message={x => x.allClaimsDashboard.messages.hasWithdrawnPartners} />
        )}

        <Acc.Renderers.Messages messages={props.messages} />

        <Acc.Section qa="current-claims-section" title={x => x.allClaimsDashboard.labels.openSectionTitle}>
          <Acc.Loader
            pending={props.currentClaims}
            render={(currentClaims, isLoading) =>
              isLoading ? (
                <Acc.Renderers.SimpleString qa="claimsLoadingMessage">
                  <Acc.Content value={x => x.allClaimsDashboard.messages.loadingClaims} />
                </Acc.Renderers.SimpleString>
              ) : (
                renderCurrentClaimsPerPeriod(currentClaims, projectDetails, partners)
              )
            }
          />
        </Acc.Section>

        <Acc.Section qa="closed-claims-section" title={x => x.allClaimsDashboard.labels.closedSectionTitle}>
          {renderPreviousClaimsSections(projectDetails, partners, previousClaims)}
        </Acc.Section>
      </Acc.Page>
    );
  };

  const renderGuidanceMessage = (isCombinationOfSBRI: boolean, partners: PartnerDto[]) => {
    // Note: we can ensure that the PM as we can check isLead as only one FC can have this
    const isCurrentOverduePartner = partners.find(x => x.isLead && x.overdueProject);

    if (!isCurrentOverduePartner && isCombinationOfSBRI) {
      return (
        <Acc.Renderers.SimpleString qa="theFinalClaimApprovedNotificationMessage">
          <Acc.Content value={x => x.allClaimsDashboard.sbriGuidanceMessage} />
        </Acc.Renderers.SimpleString>
      );
    }

    return <ClaimsDashboardGuidance overdueProject={!!isCurrentOverduePartner?.overdueProject} />;
  };

  const groupClaimsByPeriod = (claims: ClaimDto[]) => {
    const distinctPeriods = [...new Set(claims.map(x => x.periodId))].sort((a, b) => a - b);
    return distinctPeriods.map(period => {
      const periodClaims = claims.filter(x => x.periodId === period);
      return {
        periodId: period,
        claims: periodClaims,
        start: periodClaims[0].periodStartDate,
        end: periodClaims[0].periodEndDate,
      };
    });
  };

  const renderCurrentClaimsPerPeriod = (claims: ClaimDto[], project: ProjectDto, partners: PartnerDto[]) => {
    const groupedClaims = groupClaimsByPeriod(claims);
    if (groupedClaims.length === 0) {
      if (project.status === ProjectStatus.Terminated || project.status === ProjectStatus.Closed) {
        return (
          <Acc.Renderers.SimpleString qa="theFinalClaimApprovedNotificationMessage">
            <Acc.Content value={x => x.allClaimsDashboard.messages.noRemainingClaims} />
          </Acc.Renderers.SimpleString>
        );
      }

      if (!project.periodEndDate) return null;
      const date = DateTime.fromJSDate(project.periodEndDate).plus({ days: 1 }).toJSDate();
      return (
        <Acc.Renderers.SimpleString qa="notificationMessage">
          <Acc.Content value={x => x.allClaimsDashboard.messages.noOpenClaimsMessage(date)} />
        </Acc.Renderers.SimpleString>
      );
    }
    return groupedClaims.map((x, i) => renderCurrentClaims(x.periodId, x.start, x.end, x.claims, project, partners, i));
  };

  const renderCurrentClaims = (
    periodId: number,
    start: Date,
    end: Date,
    claims: ClaimDto[],
    project: ProjectDto,
    partners: PartnerDto[],
    index: number,
  ) => {
    const title = (
      <>
        Period {periodId}: <Acc.Renderers.ShortDateRange start={start} end={end} />
      </>
    );
    const ClaimTable = Acc.TypedTable<ClaimDto>();
    const renderPartnerName = (x: ClaimDto) => {
      const p = partners.filter(y => y.id === x.partnerId)[0];
      if (p) return Acc.getPartnerName(p, true);
      return null;
    };

    return (
      <Acc.Section title={title} qa="current-claims-section" key={index}>
        <ClaimTable.Table
          data={claims}
          bodyRowFlag={x => (getBodyRowFlag(x, project, partners) ? "edit" : null)}
          caption={<Acc.Content value={x => x.allClaimsDashboard.labels.openCaption} />}
          qa="current-claims-table"
        >
          <ClaimTable.Custom header={x => x.allClaimsDashboard.labels.partner} qa="partner" value={renderPartnerName} />
          <ClaimTable.Currency
            header={x => x.allClaimsDashboard.labels.forecastCosts}
            qa="forecast-cost"
            value={x => x.forecastCost}
          />
          <ClaimTable.Currency
            header={x => x.allClaimsDashboard.labels.actualCosts}
            qa="actual-cost"
            value={x => x.totalCost}
          />
          <ClaimTable.Currency
            header={x => x.allClaimsDashboard.labels.difference}
            qa="diff"
            value={x => roundCurrency(x.forecastCost - x.totalCost)}
          />
          <ClaimTable.String header={x => x.allClaimsDashboard.labels.status} qa="status" value={x => x.statusLabel} />
          <ClaimTable.ShortDate
            header={x => x.allClaimsDashboard.labels.lastUpdated}
            qa="last-update"
            value={x => x.paidDate || x.approvedDate || x.lastModifiedDate}
          />
          <ClaimTable.Custom
            header={x => x.allClaimsDashboard.labels.actionHeader}
            hideHeader
            qa="link"
            value={x => (
              <Acc.Claims.ClaimDetailsLink
                claim={x}
                project={project}
                partner={partners.find(p => p.id === x.partnerId)!}
                routes={props.routes}
              />
            )}
          />
        </ClaimTable.Table>
      </Acc.Section>
    );
  };

  const getBodyRowFlag = (claim: ClaimDto, project: ProjectDto, partners: PartnerDto[]) => {
    if (isProjectActive) return false;

    const partner = partners.find(x => x.id === claim.partnerId);
    if (!partner) return false;

    const linkType = getClaimDetailsLinkType({ claim, project, partner });

    return linkType === "edit" || linkType === "review";
  };

  const renderPreviousClaimsSections = (project: ProjectDto, partners: PartnerDto[], previousClaims: ClaimDto[]) => {
    const grouped = partners.map(x => ({ partner: x, claims: previousClaims.filter(y => y.partnerId === x.id) }));

    return (
      <Acc.Accordion qa="previous-claims">
        {grouped.map((x, i) => {
          const partnerName = getPartnerName(x.partner, true);

          return (
            <Acc.AccordionItem key={i} title={partnerName} qa={`accordion-item-${i}`}>
              {previousClaimsSection(project, x.partner, x.claims)}
            </Acc.AccordionItem>
          );
        })}
      </Acc.Accordion>
    );
  };

  const previousClaimsSection = (project: ProjectDto, partner: PartnerDto, previousClaims: ClaimDto[]) => {
    if (previousClaims.length === 0) {
      return (
        <Acc.Renderers.SimpleString qa={`noClosedClaims-${partner.accountId}`}>
          <Acc.Content value={x => x.allClaimsDashboard.messages.noClosedClaims} />
        </Acc.Renderers.SimpleString>
      );
    }
    const ClaimTable = Acc.TypedTable<ClaimDto>();
    const partnerName = getPartnerName(partner);

    return (
      <div>
        <ClaimTable.Table data={previousClaims} caption={partnerName} qa={`previousClaims-${partner.accountId}`}>
          <ClaimTable.Custom qa="period" value={x => renderClosedPeriodColumn(x)} />
          <ClaimTable.Currency
            header={x => x.allClaimsDashboard.labels.forecastCosts}
            qa="forecast-cost"
            value={x => x.forecastCost}
          />
          <ClaimTable.Currency
            header={x => x.allClaimsDashboard.labels.actualCosts}
            qa="actual-cost"
            value={x => x.totalCost}
          />
          <ClaimTable.Currency
            header={x => x.allClaimsDashboard.labels.difference}
            qa="diff"
            value={x => roundCurrency(x.forecastCost - x.totalCost)}
          />
          <ClaimTable.String header={x => x.allClaimsDashboard.labels.status} qa="status" value={x => x.statusLabel} />
          <ClaimTable.ShortDate
            header={x => x.allClaimsDashboard.labels.lastUpdated}
            qa="last-update"
            value={x => x.paidDate || x.approvedDate || x.lastModifiedDate}
          />
          <ClaimTable.Custom
            header={x => x.allClaimsDashboard.labels.actionHeader}
            hideHeader
            qa="link"
            value={x => (
              <Acc.Claims.ClaimDetailsLink claim={x} project={project} partner={partner} routes={props.routes} />
            )}
          />
        </ClaimTable.Table>
      </div>
    );
  };

  const renderClosedPeriodColumn = (claim: ClaimDto) => {
    return <Acc.Claims.ClaimPeriodDate claim={claim} />;
  };

  return (
    <Acc.PageLoader pending={combined} render={x => renderContents(x.projectDetails, x.partners, x.previousClaims)} />
  );
}

const AllClaimsDashboardContainer = (props: AllClaimsDashboardParams & BaseProps) => {
  const stores = useStores();

  return (
    <AllClaimsDashboardComponent
      {...props}
      projectDetails={stores.projects.getById(props.projectId)}
      partners={stores.partners.getPartnersForProject(props.projectId)}
      currentClaims={stores.claims.getActiveClaimsForProject(props.projectId)}
      previousClaims={stores.claims.getInactiveClaimsForProject(props.projectId)}
    />
  );
};

export const AllClaimsDashboardRoute = defineRoute({
  allowRouteInActiveAccess: true,
  routeName: "allClaimsDashboard",
  routePath: "/projects/:projectId/claims/dashboard",
  container: AllClaimsDashboardContainer,
  getParams: route => ({
    projectId: route.params.projectId ?? "",
  }),
  getTitle: ({ content }) => content.allClaimsDashboard.title(),
  accessControl: (auth, { projectId }) =>
    auth.forProject(projectId).hasAnyRoles(ProjectRole.MonitoringOfficer, ProjectRole.ProjectManager),
});
