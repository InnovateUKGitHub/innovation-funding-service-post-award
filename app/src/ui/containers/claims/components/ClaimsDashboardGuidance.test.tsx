import { render } from "@testing-library/react";

import { ProjectRole } from "@framework/constants";
import { ClaimsDashboardGuidance, ClaimsDashboardGuidanceProps } from "@ui/containers/claims/components";
import { TestBed } from "@shared/TestBed";
import { testInitialiseInternationalisation } from "@shared/testInitialiseInternationalisation";

describe("<ClaimsDashboardGuidance />", () => {
  describe("@renders", () => {
    const stubContent = {
      claimsMessages: {
        guidanceMessage: "stub-guidanceMessage",
        overdueGuidanceMessage: {
          message: "Deadline past. Too late. Email <0 /> for more information.",
          email: "webmaster@example.com",
        },
      },
    };

    const setup = (props: ClaimsDashboardGuidanceProps) => {
      return render(
        <TestBed>
          <ClaimsDashboardGuidance {...props} />
        </TestBed>,
      );
    };

    beforeAll(async () => {
      await testInitialiseInternationalisation(stubContent);
    });

    test("when overdue claim", () => {
      const { queryByTestId } = setup({ overdueProject: true });

      const guidanceMessage = queryByTestId("guidance-message");
      expect(guidanceMessage).toBeInTheDocument();
      expect(guidanceMessage).toMatchSnapshot();
    });

    describe("when not overdue", () => {
      test.each`
        name             | inputRoles                       | expectedContent
        ${"when FC"}     | ${ProjectRole.FinancialContact}  | ${stubContent.claimsMessages.guidanceMessage}
        ${"when not FC"} | ${ProjectRole.MonitoringOfficer} | ${stubContent.claimsMessages.guidanceMessage}
      `("when not overdue partner as non SBRI competion $name", ({ inputRoles, expectedContent }) => {
        const { container, queryByText } = setup({
          competitionType: "CR&D",
          roles: inputRoles,
          overdueProject: false,
        });

        expect(queryByText(expectedContent)).toBeInTheDocument();
        expect(container.querySelector(".markdown")).toBeInTheDocument();
      });

      test.each`
        name             | inputRoles                       | shouldHaveMarkdownElement | expectedContent
        ${"when FC"}     | ${ProjectRole.FinancialContact}  | ${false}                  | ${stubContent.claimsMessages.guidanceMessage}
        ${"when not FC"} | ${ProjectRole.MonitoringOfficer} | ${true}                   | ${stubContent.claimsMessages.guidanceMessage}
      `(
        "when not overdue partner as a SBRI project $name",
        ({ inputRoles, shouldHaveMarkdownElement, expectedContent }) => {
          const { container, queryByText } = setup({
            competitionType: "SBRI",
            roles: inputRoles,
            overdueProject: false,
          });

          expect(queryByText(expectedContent)).toBeInTheDocument();

          if (shouldHaveMarkdownElement) {
            expect(container.querySelector(".markdown")).toBeInTheDocument();
          } else {
            expect(container.querySelector(".markdown")).not.toBeInTheDocument();
          }
        },
      );
    });
  });
});
