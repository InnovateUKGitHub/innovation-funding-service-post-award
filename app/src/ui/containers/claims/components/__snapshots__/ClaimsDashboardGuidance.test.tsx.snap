// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClaimsDashboardGuidance /> @renders when overdue claim 1`] = `
<div
  class="govuk-warning-text-background govuk-warning-text acc-message acc-message__info"
  data-qa="guidance-message"
  data-qa-type="info"
  style="border-color: #1d70b8;"
>
  <span
    class="govuk-warning-text__assistive"
    data-qa="guidance-message-assistive"
  >
    Info
  </span>
  <div
    class="govuk-body"
    data-qa="guidance-message-content"
  >
    Deadline past. Too late. Email 
    <a
      class="govuk-link govuk-!-font-size-19"
      href="mailto:webmaster@example.com"
    >
      webmaster
      <wbr />
      @
      example.com
    </a>
     for more information.
  </div>
</div>
`;
